packages/compiler-core/src/transforms/transformExpression.ts

```diff
@@ -199,7 +199,16 @@ export function processExpression(
     } else {
       if (type && type.startsWith('setup')) {
         // setup bindings in non-inline mode
-        return `$setup.${raw}`
+        if (
+          parent &&
+          parent.type === 'CallExpression' &&
+          (type === BindingTypes.SETUP_LET || type === BindingTypes.SETUP_CONST)
```

Comment:
Why check `type` here? Consider [this example](https://deploy-preview-6827--vue-sfc-playground.netlify.app/#eNqlUkFOwzAQ/IrlC61a24gDoCpF8A9fUmfTGhLbsjctUZS/s06iqsAFiVt2ZzyzO9mBv4Ugzx3wHS+SiTYgS4BdeNHOtsFHZAOLULOR1dG37I6odzdQ7f0VkgohocREBO2MdwmZ8Z1Dts8Sq/u1doWaTUieCoQ2NCUCVYwVlT0z05Qp7TU3Zaw0n/qEHDpE7xj2AQibK83Zq2ms+cjs7LLZ0IPZzyY2DIv3OBZqfvFXNdpptSYt32HolhW+aRSKRqWvgha+LsC3fE5FtGWQ78k7inTIdL0ASfMdmzq5R0HmWvMTYkg7pSoIje9FiHC2cBGPzw9PQhBLpNoIcumPkSappANsbN3LMgRFsIw0nm1BQmrFIfpLgkj2mm9vnBQ1zxBFBFdBhPh/5x+Cv9yz+ajdSLEsV0FxwOd0NnXnDFr6BVPScyT5XHwDsvHHFZ5sklPsdDEjH78AdCvswQ==), the problem consists

Reply:
1. I had to make sure it's a function identifier with `parents.type`. Then use `.bind()` to change the scope to `undefined`.

Reply:
There are many kinds of `BindingTypes`. For generality, I don't think it's necessary to check `BindingTypes` here.

Or if you have a better solution to fix the example I provided, it's also all right.

Reply:
Updated! The previous judgment was poor, and the following conditions are also met.
```js
const foo = ref(function(a, b) {
  console.log(this)
})
```

Reply:
Great, but CI is failed. You can push an empty commit `git commit --allow-empty -m "Empty-Commit"` to re-run CI.